tasks.c

```diff
@@ -1118,16 +1118,24 @@ static void prvAddNewTaskToReadyList( TCB_t * pxNewTCB ) PRIVILEGED_FUNCTION;
 
                     if( ( uxCoreMap & ( ( UBaseType_t ) 1U << ( UBaseType_t ) xCoreID ) ) != 0U )
                     {
-                        /* The ready task that was removed from this core is not excluded from it.
-                         * Only look at the intersection of the cores the removed task is allowed to run
-                         * on with the cores that the new task is excluded from. It is possible that the
-                         * new task was only placed onto this core because it is excluded from another.
-                         * Check to see if the previous task could run on one of those cores. */
+                        /* pxPreviousTCB was removed from this core which is not excluded from it's
+                         * own core affinity mask. Therefore, it is preempted by the new higher
+                         * priority task, the pxCurrentTCBs[ xCoreID ]. Only look at the
+                         * intersection of the cores on which pxPreviousTCB is allowed to run and
+                         * the cores on which pxCurrentTCBs[ xCoreID ] is not allowed to run. This
+                         * reduce the number of cores need to be searched for pxPreviousTCB to run.
```

Comment:
nit: small changes to the comment
  *This reduces the number of cores needed to be searched for pxPreviousTCB to run.*